src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/SearchParameterState/SearchParameterStateHandler.cs

```diff
@@ -0,0 +1,150 @@
+using Microsoft.Health.Fhir.Core.Features.Security;
+using Microsoft.Health.Fhir.Core.Messages.SearchParameterState;
+using Microsoft.Health.Fhir.Core.Models;
+using static Hl7.Fhir.Model.Parameters;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.SearchParameterState
+{
+    public class SearchParameterStateHandler : IRequestHandler<SearchParameterStateRequest, SearchParameterStateResponse>
+    {
+        private readonly IAuthorizationService<DataActions> _authorizationService;
+        private readonly ISearchParameterDefinitionManager _searchParameterDefinitionManager;
+        private readonly SearchParameterStatusManager _searchParameterStatusManager;
+
+        public SearchParameterStateHandler(IAuthorizationService<DataActions> authorizationService, ISearchParameterDefinitionManager searchParameterDefinitionManager, SearchParameterStatusManager searchParameterStatusManager)
+        {
+            EnsureArg.IsNotNull(authorizationService, nameof(authorizationService));
+            EnsureArg.IsNotNull(searchParameterDefinitionManager, nameof(searchParameterDefinitionManager));
+            EnsureArg.IsNotNull(searchParameterStatusManager, nameof(searchParameterStatusManager));
+
+            _authorizationService = authorizationService;
+            _searchParameterDefinitionManager = searchParameterDefinitionManager;
+            _searchParameterStatusManager = searchParameterStatusManager;
+        }
+
+        public async Task<SearchParameterStateResponse> Handle(SearchParameterStateRequest request, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(request, nameof(request));
+
+            if (await _authorizationService.CheckAccess(DataActions.Read, cancellationToken) != DataActions.Read)
+            {
+                throw new UnauthorizedFhirActionException();
+            }
+
+            IEnumerable<SearchParameterInfo> searchParameterResult = new List<SearchParameterInfo>();
+            if (request.Queries.Count == 0)
+            {
+                searchParameterResult = _searchParameterDefinitionManager.AllSearchParameters;
+            }
+            else
+            {
+                List<IEnumerable<SearchParameterInfo>> results = new List<IEnumerable<SearchParameterInfo>>();
+                foreach (Tuple<string, string> query in request.Queries)
+                {
+                    string[] queryValues = query.Item2.Split(',');
+
+                    switch (query.Item1.ToLowerInvariant())
+                    {
+                        case SearchParameterStateProperties.ResourceType:
+                            results.Add(_searchParameterDefinitionManager.GetSearchParametersByResourceTypes(queryValues).DefaultIfEmpty());
+                            break;
+                        case SearchParameterStateProperties.Code:
+                            results.Add(_searchParameterDefinitionManager.GetSearchParametersByCodes(queryValues).DefaultIfEmpty());
```

Comment:
These Add statements appear to be building a collection of all SearchParameters that match any of the search properties passed in by the user.  I.E. if a query string was:  <fhirUrl>/SearchParameter/$status?url=http://foo&code=identifier,  this implementation would return all SearchParameter status that have a url of http://foo OR code=identifier.  What should be returned is the list of SearchParameter status that a url of http://foo AND code=identifier.  I believe it should be an intersect method instead of Add.

Reply:
Correct. I'm gathering each of the lists and at the end I created a method that intersects them before returning.

Reply:
Yes!  Sorry I missed that.  Thanks.